sui/src/sui_commands.rs

```diff
@@ -29,53 +30,105 @@ use crate::config::{
             }
-            SuiCommand::Genesis {
-                working_dir,
-                config: path,
-            } => {
-                let network_path = working_dir.join("network.conf");
-                let wallet_path = working_dir.join("wallet.conf");
-                let keystore_path = working_dir.join("wallet.key");
-                let db_folder_path = working_dir.join("client_db");
-
-                if let Ok(config) = PersistedConfig::<NetworkConfig>::read(&network_path) {
-                    if !config.authorities.is_empty() {
-                        return Err(anyhow!("Cannot run genesis on a existing network, please delete network config file and try again."));
+            SuiCommand::Genesis { working_dir, force } => {
+                let sui_config_dir = &match working_dir {
+                    // if a directory is specified, it must exist (it
+                    // will not be created)
+                    Some(v) => v.clone(),
+                    // create default Sui config dir if not specified
+                    // on the command line and if it does not exist
+                    // yet
+                    None => {
+                        let config_path = sui_config_dir()?;
+                        fs::create_dir_all(&config_path)?;
+                        config_path
                     }
-                }
-                let genesis_conf = if let Some(path) = path {
-                    PersistedConfig::read(path)?
-                } else {
-                    GenesisConfig::default_genesis(working_dir)?
                 };
+
+                // if Sui config dir is not empty then either clean it
+                // up (if --force/-f option was specified or report an
+                // error
+                if sui_config_dir
+                    .read_dir()
+                    .map_err(|err| {
+                        anyhow!(err)
+                            .context(format!("Cannot open Sui config dir {:?}", sui_config_dir))
+                    })?
+                    .next()
+                    .is_some()
+                {
+                    if *force {
+                        fs::remove_dir_all(sui_config_dir).map_err(|err| {
+                            anyhow!(err).context(format!(
+                                "Cannot remove Sui config dir {:?}",
+                                sui_config_dir
+                            ))
+                        })?;
+                        fs::create_dir(sui_config_dir).map_err(|err| {
+                            anyhow!(err).context(format!(
+                                "Cannot create Sui config dir {:?}",
+                                sui_config_dir
+                            ))
+                        })?;
+                    } else {
+                        bail!("Cannot run genesis with non-empty Sui config directory {}, please use --force/-f option to remove existing configuration", sui_config_dir.to_str().unwrap());
```

Comment:
Presumably, we could also overwrite the existing configuration ourselves and proceed, since this is what we are asking the user to do manually. Is there a risk they will clobber useful data by doing this?

Reply:
> Presumably, we could also overwrite the existing configuration ourselves and proceed, since this is what we are asking the user to do manually. Is there a risk they will clobber useful data by doing this?

I did not want to make this "silent" as the docs describe how to change the config manually and having a manually modified accidentally removed could be annoying. We can certainly revisit this decision, though!

Reply:
Fair point about clobbering the manual configs. Maybe there is a middle ground like automated deletion with a prompt ("About to run genesis. This will clobber any manual changes to configs. Proceed?" Y/N).